File: src/main/java/com/github/daniel/shuy/kotlin/spring/boot/example/model/Pet.java
package com.github.daniel.shuy.kotlin.spring.boot.example.model;

import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class Pet {
  @Id
  @GeneratedValue
  private Long id;

  private String name;

  @Enumerated(EnumType.STRING)
  private Status status;

  @ElementCollection(fetch = FetchType.LAZY)
  @Column(name = "tag")
  private Set<String> tags;
}

File: src/main/kotlin/com/github/daniel/shuy/kotlin/spring/boot/example/model/Pet.kt
package com.github.daniel.shuy.kotlin.spring.boot.example.model

import jakarta.persistence.Column
import jakarta.persistence.ElementCollection
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id

@Entity
class Pet(
    var name: String,

    @Enumerated(EnumType.STRING)
    var status: Status,

    @ElementCollection(fetch = FetchType.LAZY)
    @Column(name = "tag")
    var tags: Set<String>? = null,
) {
    @Id
    @GeneratedValue
    var id: Long? = null
}
